pkg/asset/imagebased/image/baseiso.go

```diff
@@ -0,0 +1,140 @@
+package image
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/coreos/stream-metadata-go/arch"
+	"github.com/coreos/stream-metadata-go/stream"
+	"github.com/sirupsen/logrus"
+
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/rhcos"
+	"github.com/openshift/installer/pkg/rhcos/cache"
+	"github.com/openshift/installer/pkg/types"
+)
+
+var (
+	baseIsoFilename = ""
+
+	defaultCoreOSStreamGetter = rhcos.FetchCoreOSBuild
+)
+
+// BaseIso generates the base ISO file for the image.
+type BaseIso struct {
+	File *asset.File
+
+	streamGetter CoreOSBuildFetcher
+}
+
+// CoreOSBuildFetcher will be to used to switch the source of the coreos metadata.
+type CoreOSBuildFetcher func(ctx context.Context) (*stream.Stream, error)
+
+var _ asset.WritableAsset = (*BaseIso)(nil)
+
+// Name returns the human-friendly name of the asset.
+func (i *BaseIso) Name() string {
+	return "BaseIso Image"
+}
+
+// Dependencies returns dependencies used by the asset.
+func (i *BaseIso) Dependencies() []asset.Asset {
+	return []asset.Asset{}
+}
+
+// Generate generates the base ISO.
+func (i *BaseIso) Generate(dependencies asset.Parents) error {
+	var err error
+	var baseIsoFileName string
+
+	if urlOverride, ok := os.LookupEnv("OPENSHIFT_INSTALL_OS_IMAGE_OVERRIDE"); ok && urlOverride != "" {
+		logrus.Warn("Found override for OS Image. Please be warned, this is not advised.")
+		baseIsoFileName, err = cache.DownloadImageFile(urlOverride, cache.ImageBasedApplicationName)
+	} else {
+		if i.streamGetter == nil {
+			i.streamGetter = defaultCoreOSStreamGetter
+		}
+		baseIsoFileName, err = i.downloadBaseIso(arch.RpmArch(types.ArchitectureAMD64))
```

Comment:
Consider getting the architecture from the user

Reply:
I wasn't sure if I should go for this or not, I'll try to implement it :+1:

Reply:
Are we building lca for multiarch?

Reply:
IIUC the `lca-cli` is currently [built](https://github.com/openshift-kni/lifecycle-agent/blob/main/Dockerfile#L23) and [released](https://github.com/openshift/release/blob/master/ci-operator/config/openshift-kni/lifecycle-agent/openshift-kni-lifecycle-agent-release-4.16.yaml#L8-L11) only for `amd64`. 

I added a comment above this line to indicate/justify the default `amd64` architecture choice. WDYT?

Reply:
I think we should open ticket for LCA to build multi arch image but till than there is no point to support anything rather than amd64.